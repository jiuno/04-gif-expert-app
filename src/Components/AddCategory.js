import React, { useState } from 'react';
import PropTypes from 'prop-types';


export  const AddCategory = ({SetCategories}) => {
    
        const  [InputValue, setInputValue] = useState('');

        const handleInputChange = (e) => {
            
            setInputValue(e.target.value)

            console.log('HandleInputChange llamado');

        }
        
        const handleSubmit = (e) => {
            e.preventDefault();
            if (InputValue.trim().length >= 2) {
                SetCategories((cats) => [InputValue, ...cats]); //Esto es categories ahora
                setInputValue(''); 
                console.log('HandleSubmit llamado')
                
            }


            // console.log('Submit hecho');

        }
        return (
            <form onSubmit= {handleSubmit}>
                <h3>Vas a enviar { InputValue }</h3> {/*Esto es temporal*/}
                <input
                type="text" 
                value={InputValue}
                onChange={handleInputChange}
                
                />
            </form>
        )
    }

AddCategory.propTypes = {
    SetCategories: PropTypes.func.isRequired
}

