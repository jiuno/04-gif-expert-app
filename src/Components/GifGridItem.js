import React from 'react'

export const GifGridItem = ({id, title, url}) => {
    
    // console.log(url);
    
    return (
        <div className="card animate__animated animate__bounceIn ">
            <img src={url} alt="GIF" ></img>
            <hr></hr>
            <h3> {title} </h3>
        </div>
    )
}
