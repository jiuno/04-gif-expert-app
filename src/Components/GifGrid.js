import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs( category );

    return (
        <>
        <h3> {category} </h3>

        <div className="Card-Grid">
           
                {loading && <p className="animate__animated animate__heartBeat">Loading...</p> }
                { images.map( (img) =>  (
                <GifGridItem 
                { ...img} //Manda props.img con todas sus propiedades, no solo id, title y url.
                key={img.id}
                />))}

        </div>
        </>
    )
}
