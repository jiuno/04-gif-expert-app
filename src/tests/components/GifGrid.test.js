import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from '../../Components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas para el componente <GifGrid />', () => {
    
    test('debe matchear con el snapshot', () => {

        const category = 'un string'; 
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        
        expect(wrapper).toMatchSnapshot(); 

    })

    test('debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        
        const category = 'un string'; 

        const gifs = [{
            id: 'Hola',
            url: 'https://localhost/sale/pimba.jpg',
            title: 'Tuqui'
        },{
            id:'Que tal',
            url: 'https://localhost/sale/pimba.jpg',
            title: 'Tuqui'    
        }]


        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        // expect(wrapper).toMatchSnapshot();
        
        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


       
        

    })
    
    

})
