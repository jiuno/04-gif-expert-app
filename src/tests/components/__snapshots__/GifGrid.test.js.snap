// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifGrid /> debe matchear con el snapshot 1`] = `
<Fragment>
  <h3>
     
    un string
     
  </h3>
  <div
    className="Card-Grid"
  >
    <p
      className="animate__animated animate__heartBeat"
    >
      Loading...
    </p>
  </div>
</Fragment>
`;
